### PR TITLE
Update contributing docs

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

To kick FerretDB/engineering#63 forward.

## Readiness checklist

- [ ] I added/updated unit tests.
- [ ] I added/updated integration/compatibility tests.
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -220,6 +221,18 @@ Some of our idiosyncrasies:
    It may seem random, but it is only pseudo-random and follows BSON spec: <https://bsonspec.org/spec.html>
 2. We generally pass and return `struct`s by pointers.
    There are some exceptions like `types.Path` that has value semantics, but when in doubt – use pointers.
+3. Code comments:
+   - All top-level declarations, even unexpected, should have documentation comments.
```
Comment (chilagrow) [inline CONTRIBUTING.md:225]: What is unexpected in this context?

Comment (AlekSi) [reply CONTRIBUTING.md:225]: Autocorrect :) It should be unexported

Comment (chilagrow) [reply CONTRIBUTING.md]: Ah that makes sense :see_no_evil:

**Review by chilagrow — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -220,6 +221,18 @@ Some of our idiosyncrasies:
...
 2. We generally pass and return `struct`s by pointers.
    There are some exceptions like `types.Path` that has value semantics, but when in doubt – use pointers.
+3. Code comments:
+   - All top-level declarations, even unexported, should have documentation comments.
+   - In documentation comments do not describe the name in terms of the name itself (`// Registry is a registry of …`).
+     Use other words instead; often, they could add additional information and make reading more pleasant (`// Registry stores …`).
+   - In code comments, in general, do not describe _what_ the code does: it should be clear from the code itself
+     (and when it doesn't and the code is tricky, simplify it instead).
+     Instead, describe _why_ the code does that if it is not clear from the surrounding context, names, etc.
+     There is no need to add comments just because there are no comments if everything is already clear without them.
```
Comment (noisersup) [inline CONTRIBUTING.md:231]: It would be nice to provide some example here in the future

**Review by noisersup — APPROVED** [review]
LGTM, one small comment for the future consideration

**Pull request merged**